### ISSUE TITLE
[Question] Does OpenSearch offer version control

### ISSUE THREAD

**Issue by collinsezedike** [opensearch-project/OpenSearch#10145]
According to the [documentation](https://opensearch.org/docs/latest/about/#indices-and-documents), when you add a document to an index, OpenSearch adds some metadata. One of these data is _version. I have the following questions:
1. Is `_version` used for version control?  
2.  If yes, how does OpenSearch manage the different versions without convoluting the index?  
3. If yes, can versions be queried and visualized? For example, say I use OpenSearch to catalogue blog posts. And I typically use OpenSearch to document blog posts' metrics. Could I also use OpenSearch to see how much I rewrite each blog post?

**Comment by msfroh** [comment]
1. Not exactly -- only the latest version is kept. If you specify the `version` parameter when indexing a document, it will check against the internal version of the existing doc and only overwrite it if the existing version matches (as a way of doing optimistic locking).
2. Not applicable.
3. You can't get the old versions (unless you choose to write a new doc for every version, but then you'd need to find a way to show the *latest* version on your blog -- maybe using `collapse`). If you `GET /<index_name>/_doc/<doc_id>`, then you can see the current version. That would at least tell you how many times you've edited a given blog post (but not what edits were made).

Hope that helps! Feel free to follow up with any further questions.

**Comment by msfroh** [comment]
Oh, not directly related to your questions, but it's also worth noting that you can use external versioning (by specifying `version_type=external`) if you have some outside system that assigns increasing versions, so that if multiple sources are updating the same doc, then you can make sure the last update always wins.

**Comment by msfroh** [comment]
@collinsezedike -- Did this cover all your questions? If so, please feel free to close it.

If not, it may still be worthwhile from a SEO standpoint to close out this question and ask new questions on a new issue. 

I know I've certainly run into the case where I've tried to Google "How do I do X?" and get no answer, but I've tried to Google "How do I do X but not Y", and land on a page where someone asked "How do I do Y?" and someone told them how to do X.

tl;dr: if you have more questions related to your original problem, please ask them here. If you have more questions not related to your original problem, please open a new issue.

If you want general advice on implementing a blog (especially if you're considering OpenSearch as a platform), feel free to DM me. I implemented a blog site back when I was a student (back when PHP + MySQL was the norm), so I may have useful but potentially out-of-date advice.

**Comment by collinsezedike** [comment]
Yes it does, @msfroh.
Thank you.
Closing now...
